operational/azure/vms_without_managed_disks/azure_vms_without_managed_disks.pt

```diff
@@ -0,0 +1,147 @@
+    additional_params do {
+      "resource" => "https://management.azure.com/"
+    } end
+  end
+end
+
+###############################################################################
+# Pagination
+###############################################################################
+
+pagination "azure_pagination" do
+  get_page_marker do
+    body_path "nextLink"
+  end
+  set_page_marker do
+    uri true
+  end
+end
+
+###############################################################################
+# Datasources
+###############################################################################
+
+datasource "ds_azure_virtual_machines" do
+  request do
+    auth $azure_auth
+    pagination $azure_pagination
+    host "management.azure.com"
+    path join(["/subscriptions/", $param_azure_sub, "/providers/Microsoft.Compute/virtualMachines"])
+    query "api-version","2018-06-01"
+    header "User-Agent", "RS Policies"
+  end
+  result do
+    encoding "json"
+    collect jmes_path(response, "value[*]") do
+      field "id", jmes_path(col_item,"id")
+      field "type", jmes_path(col_item,"type")
+      field "name", jmes_path(col_item,"name")
+      field "osDisk", jmes_path(col_item,"properties.storageProfile.osDisk")
+    end
+  end
+end
+
+datasource "ds_filtered_virtual_machines" do
+  run_script $js_filter_virtual_machines, $ds_azure_virtual_machines
+end
+
+###############################################################################
+# Scripts
+###############################################################################
+script "js_filter_virtual_machines", type: "javascript" do
+  parameters "ds_azure_virtual_machines"
+  result "results"
+  code <<-EOS
+    var results = [];
+    _.each(ds_azure_virtual_machines, function(vm){
+      if (vm.osDisk.vhd != null && vm.osDisk.vhd !== undefined){
+        results.push(vm)
+      }
+    })
+EOS
+end
+###############################################################################
+# Policy
+###############################################################################
+
+policy "azure_resource_policy" do
+  validate_each $ds_filtered_virtual_machines do
+    summary_template "{{ rs_project_name }} (Account ID: {{ rs_project_id }}): {{ len data }} Azure Resources out of compliance"
+    detail_template <<-EOS
+# Azure Resources out of compliance
+| Resource ID | osDisk VHD |
+| ----------- | ---------- |
+{{ range data -}}
+| {{.id}} | {{.osDisk.vhd.uri}} |
+{{ end -}}
+EOS
+    escalate $email
+    check eq(0,1)
+  end
+end
+
+###############################################################################
+# Escalations
+###############################################################################
+
+escalation "email" do
+   email $param_email
+end
+
+###############################################################################
+# Cloud Workflow
+###############################################################################
+
+define sys_log($subject, $detail) do
```

Comment:
No CWF in this policy

Reply:
I feel like we should keep sys_log in there for future enhancements. I think of it as part of the default template